docs/book/src/topics/aso.md

```diff
@@ -0,0 +1,55 @@
+# Azure Service Operator
+
+## Overview
+
+CAPZ interfaces with Azure to create and manage some types of resources using [Azure Service Operator
+(ASO)](https://azure.github.io/azure-service-operator/).
+
+More context around the decision for CAPZ to pivot towards using ASO can be found in the
+[proposal](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/docs/proposals/20230123-azure-service-operator.md).
+
+## Primary changes
+
+For most users, the introduction of ASO is expected to be fully transparent and backwards compatible. Changes
+that may affect specific use cases are described below.
+
+### Installation
+
+Beginning with CAPZ v1.11.0, ASO's control plane will be installed automatically by `clusterctl` in the
```

Comment:
Is this a permanent behavior or at some future date BYO ASO might be supported?

Reply:
We haven't yet discussed any plans to support BYO ASO. I think the main roadblock to that is that `clusterctl` expects certain labels applied to the CRDs for things like `clusterctl move` to work. In theory, if users add those themselves to the ASO CRDs, then things should work, but that would require a fair bit of testing that we haven't done yet.

And I think we would need `clusterctl init` to support some conditional logic that could enable/disable the ASO install alongside CAPZ to support existing users relying on ASO being installed that way.

Reply:
Do we have an issue to discuss BYO ASO?

Reply:
I'm not necessarily saying you need one until such time as a user asks about it - just something worth thinking about ahead of time as it is possible it will come up.

Reply:
Opened #3900 to track